arch/common/CMakeLists.txt

```diff
@@ -32,7 +32,7 @@ zephyr_linker_sources_ifdef(CONFIG_GEN_IRQ_VECTOR_TABLE
 )
 
 if(CONFIG_GEN_ISR_TABLES)
-  zephyr_linker_section(NAME .intList VMA IDT_LIST LMA IDT_LIST NOINPUT PASS NOT LINKER_ZEPHYR_FINAL)
+  zephyr_linker_section(NAME .intList NOINPUT TYPE (INFO) PASS NOT LINKER_ZEPHYR_FINAL)
```

Comment:
this will not work.
The type specified here is not a GNU ld type, but a generic type which then maps to the linker script depending on whether that is ld or a scatter file.

That is also why you see a CMake warning with this PR:
```
CMake Warning at /projects/github/ncs/zephyr/cmake/modules/extensions.cmake:4247 (message):
  zephyr_linker_section(NAME ...) given unknown arguments: INFO;)                    
```

ld:
https://github.com/zephyrproject-rtos/zephyr/blob/b5bfe0a264b648224bc8c49d993c648fd7e43be8/cmake/linker/ld/ld_script.cmake#L192-L193

scatter:
https://github.com/zephyrproject-rtos/zephyr/blob/b5bfe0a264b648224bc8c49d993c648fd7e43be8/cmake/linker/armlink/scatter_script.cmake#L72
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker/armlink/scatter_script.cmake#L269

so for this to work with the linker generator then we need a new type there.

INFO would be a good name 😉 

See also: https://github.com/zephyrproject-rtos/zephyr/issues/61846